### PR TITLE
update truth variable summary catalog and reader (with hdf5 file)

### PR DESCRIPTION
**Pull request by danielsf**
This should address the problem identified here

https://github.com/LSSTDESC/gcr-catalogs/issues/272

specifically, that there is no reliable way to get truth redshifts for SNe in run 1.2

### CODE REVIEW

**Comment by danielsf** [comment]
Okay.  I took the native filter implementation out and added a warning if the user tries to use native filters.

File: GCRCatalogs/dc2_truth.py [hunk 1]
```diff
@@ -1,10 +1,75 @@
...
+
+    Parameters
+    ----------
+    filename: str
+        path to the hdf5 file containing the summary catalog
+    """
+
+    def _subclass_init(self, **kwargs):
+        self._file_name = kwargs['filename']
+        self._native_filter_quantities = {'redshift', 'agn', 'sn', 'sprinkled'}
```
Comment (yymao) [inline GCRCatalogs/dc2_truth.py]:
```suggestion
```

File: GCRCatalogs/dc2_truth.py [hunk 2]
```diff
@@ -1,10 +1,75 @@
 import os
 import sqlite3
 import numpy as np
+import h5py
+import warnings
 from GCR import BaseGenericCatalog
+from GCR import GCRQuery
```
Comment (yymao) [inline GCRCatalogs/dc2_truth.py]:
```suggestion
```

File: GCRCatalogs/dc2_truth.py [hunk 3]
```diff
@@ -1,10 +1,75 @@
...
+                  'all other objects'}
+
+        self._info_dict['sprinkled'] = {'units': 'unitless',
+                   'description': 'an int that is 1 if the object was '
+                   'added by the sprinkler; 0 otherwise.'}
+
+    def get_quantity_info(self, qty_name):
+        if qty_name not in self._info_dict:
+            return None
+        return self._info_dict[qty_name]
```
Comment (yymao) [inline GCRCatalogs/dc2_truth.py]: Lines 56-59: overwrite the private method instead:
```python
    def _get_quantity_info_dict(self, quantity, default=None):
		return self._info_dict.get(quantity, default)
```

**Comment by danielsf** [comment]
I have implemented your requested changes.

File: GCRCatalogs/dc2_truth.py [hunk 4]
```diff
@@ -1,10 +1,71 @@
...
+
+        self._info_dict['sn'] = {'units': 'unitless',
+                  'description': 'an int that is 1 for supernovae and 0 for '
+                  'all other objects'}
+
+        self._info_dict['sprinkled'] = {'units': 'unitless',
+                   'description': 'an int that is 1 if the object was '
+                   'added by the sprinkler; 0 otherwise.'}
+
+    def _get_quantity_info(self, qty_name, default=None):
```
Comment (yymao) [inline GCRCatalogs/dc2_truth.py]: @danielsf sorry, CI spotted a typo here. 
```suggestion
    def _get_quantity_info(self, quantity, default=None):
```

**Pull request merged**